### PR TITLE
cmake: Build in release mode

### LINKED ISSUE #21155: CMake formula builds in Debug mode?
# Please always follow these steps:
- [does not apply] Confirmed this is a problem with `brew install`ing one, specific Homebrew/homebrew-core formula (not cask or tap) and not every time you run `brew`? If it's a general `brew` problem please file this issue at https://github.com/Homebrew/brew/issues/new. If it's a `brew cask` problem please file this issue at https://github.com/caskroom/homebrew-cask/issues/new. If it's a tap (e.g. Homebrew/homebrew-php) problem please file this issue at the tap.
- [does  not apply] Ran `brew update` and retried your prior step?
- [does not apply ] Ran `brew doctor`, fixed all issues and retried your prior step?
- [does not apply] Ran `brew gist-logs <formula>` (where `<formula>` is the name of the formula that failed) and included the output link?
- [does not apply] If `brew gist-logs` didn't work: ran `brew config` and `brew doctor` and included their output with your issue?

To help us debug your issue please explain:
- What you were trying to do (and why)
Using cmake from brew
- What happened (include command output)
It worked but cmake seems very slow compared to my Ubuntu install
- What you expected to happen
Faster cmake, built with optimizations turned on.
- Step-by-step reproduction instructions (by running `brew install` commands)
brew install cmake

# Formula additions or changes
I am willing to contribute a pull request. Yet, I first wanted to confirm my suspicion:
Is the the cmake formula building cmake without optimizations turned on? I only see standard build system invocations for building cmake and no special flags or the CMAKE_BUILD_TYPE being set to Release. This usually yields a binary with optimizations turned off. Please confirm or refute my hypothesis.

### PR DESCRIPTION
**Pull request by kaihowl**
Fixes #21155

- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

**Comment by ilovezfs** [comment]
have you benchmarked any difference here with the new build itself vs. the old build. or are we just assuming this works?

**Comment by kaihowl** [comment]
Yes, I did benchmark the difference on a larger, private project. This project has roughly 2000 CMakeLists.txt files and roughly 8700 source files. (Not sure what factors more directly influence the running time of CMake.)

Based on the bottled version of 75a1f7d4f0:
initial run: cmake ~/git/<projectName>  53.57s user 19.47s system 30% cpu 3:59.86 total
rerun: cmake ~/git/<projectName> 49.81s user 19.64s system 39% cpu 2:57.82 total

Based on the version built from source of d9ffe42fa7:
cmake ~/git/<projectName> 53.20s user 17.09s system 36% cpu 3:11.40 total
rerun cmake ~/git/<projectName> 49.66s user 18.08s system 40% cpu 2:48.58 total

In short: There is some smaller difference in run time. Yet, the process seems mostly IO bound. I re-measured a couple times but the numbers have not changed significantly. I believe it is a good idea to ship software with optimizations turned on. Moreover, these numbers might change with future release of CMake.

The final choice is of course up to you. I would fully understand if you do not find the numbers convincing enough to take in the change.

**Comment by ilovezfs** [comment]
Thanks for the pull request @kaihowl! Shipped.
